### DISCUSSION TITLE
How to properly set bounds for reused map

### DISCUSSION THREAD

**Discussion by shishkin** [uber/react-map-gl#1914] [Q&A]
I have a Next.js app with a page that has a map with `reuseMaps={true}` and `initialViewState={{ bounds }}`. The reason to reuse maps is mentioned cost and potential for users to navigate between the pages and Next.js manage the state transitions with the same map instance. However, each page needs to show a map with different region. I use `initialViewState` because it takes bounds directly instead of a center and sizes (I get bbox of all GeoJSON features I need to show and use that for bounds).

I have a sporadic issue with this setup that sometimes the map will center on a region used in a previous page. That happens after wildly clicking back and forth between pages and also using browser back button. But it's not consistently reproducible.

Now I wonder what is the best practice to set initial page-dependent view state to a map that is reused across pages.

I tried `useEffect` to call `fitBounds`, but the `mapRef` is not available at that point and is `null`. I tried to call `fitBounds` inside `onLoad` callback and that resulted in a weird visible jump, where the map showed 0,0 first and then jumped to desired region.

And yes, I read about controlled state management http://visgl.github.io/react-map-gl/docs/get-started/state-management#controlled-map. As mentioned I want to just set bounds, not center and zoom.

**Comment by Pessimistress** [comment]
I don't think the current code handles `initialViewState.bounds` when a map is reused. Can you open a bug for this?

**Reply by shishkin** [reply]
Thanks for acknowledgment. I've raised #1921.
